Docs async switch

danmaz74 opened the pull request:
I made the changes we discussed about the order of async and middleware; I also tried to simplify some points. I'm not 100% satisfied with the results, but don't have time to do better. Hopefully, I updated all cross references, but if you remember more of them, please update those yourself.

gaearon commented on docs/advanced/AsyncActions.md:
```diff
@@ -293,79 +293,16 @@ Remember that reducers are just functions, so you can use functional composition
-
-Clearly, actions can’t express control flow. The best tool for control flow is a function. A function can have an `if` statement, or an early `return`. If a function has access to a `dispatch` method, it can call it many times, potentially asynchronously. Does this ring a bell?
+Finally, how do we use the synchronous action creators we [defined earlier](#synchronous-action-creators) together with network requests? The standard
+ way to do it with Redux is to use the redux-thunk middleware. We'll explain how middleware works in general [later](Middleware.md); for now, there 
```
Let's make `redux-thunk` a link to the project.

gaearon commented on docs/advanced/AsyncActions.md:
```diff
@@ -293,79 +293,16 @@ Remember that reducers are just functions, so you can use functional composition
+ is just one important thing you need to know: by using this specific middleware, an action creator can return a function instead an action object.
+ This way, the function creator becomes a [thunk](https://en.wikipedia.org/wiki/Thunk).
+ 
+When a function creator returns a function, that function will get executed by the redux-thunk middleware. This function doesn't need to be pure; 
```
I'd say keep `redux-thunk` a link here too, it looks weird in a wall of text.

gaearon commented on docs/advanced/AsyncActions.md:
````diff
@@ -293,79 +293,16 @@ Remember that reducers are just functions, so you can use functional composition
-```
-
-If this doesn’t make sense, you need to go back to the [middleware introduction](Middleware.md). This lets us rewrite our example so that `fetchPosts()` is just another action creator, but it returns a function:
+We can still define these special thunk action creators inside our action.js file:
````
Should be `actions.js` (and with backticks so it looks nicer)

gaearon commented:
This is great, thank you so much.
I left a few nitpicky comments, but it's good to go.
Let me know if you don't have time to fix those—I'll do that myself in this case.

:+1:

danmaz74 commented:
Glad you like it; it's an itch I really wanted to scratch. I'll do those small updates later today or tomorrow.  :+1:

danmaz74 commented:
I added a link to redux-thunk and fixed actions.js with backticks. One more link just a few lines below looked like too much me, but if you still want it you can add :)

gaearon commented on docs/advanced/AsyncActions.md:
```diff
@@ -293,79 +293,16 @@ Remember that reducers are just functions, so you can use functional composition
-
-Clearly, actions can’t express control flow. The best tool for control flow is a function. A function can have an `if` statement, or an early `return`. If a function has access to a `dispatch` method, it can call it many times, potentially asynchronously. Does this ring a bell?
+Finally, how do we use the synchronous action creators we [defined earlier](#synchronous-action-creators) together with network requests? The standard
+ way to do it with Redux is to use the [edux-thunk middleware](https://github.com/gaearon/redux-thunk). We'll explain how middleware works in general [later](Middleware.md); for now, there 
```
I think you meant "redux", not "edux" :-)

gaearon commented on docs/advanced/AsyncActions.md:
```diff
@@ -293,79 +293,16 @@ Remember that reducers are just functions, so you can use functional composition
+ is just one important thing you need to know: by using this specific middleware, an action creator can return a function instead an action object.
+ This way, the function creator becomes a [thunk](https://en.wikipedia.org/wiki/Thunk).
+ 
+When a function creator returns a function, that function will get executed by the redux-thunk middleware. This function doesn't need to be pure; 
```
I just think "redux-thunk middleware" doesn't look nicely in text. IMO it should either be a link, or "Redux Thunk middleware".

danmaz74 commented:
Updated again...

Anyway, didn't you hear about edux? It's the next big thing in Javascript frameworks... :D

danmaz74 commented:
@gaearon I see that now Github says there are conflicts, did you do any changes to the docs on master?

gaearon commented:
With minor tweaks, this is in, thank you so much!

danmaz74 commented:
Glad it helped :)
